docs/SECURITY.md

```diff
@@ -0,0 +1,314 @@
+servers achieve [offsets of under a second over the globe](https://www.eecis.udel.edu/~mills/database/reports/ntp-survey99-minar.pdf)). 
+
+**Broadcast channel**: The randomness generation models only needs a regular
+broadcast channel. It does **not** need to be reliable given the deterministic
+nature of a partial beacon and the drand chain: for a given round, a given node
+can only send one valid partial beacon.
+
+**Threshold**: The threshold is the amount of nodes that must be online and
+honest at a given time to broadcast their partial signature in order to create
+the final random beacon. See the attack section to know what are the
+consequences when that is not the case.
+
+**Determinism of the chain**: The chain is deterministic with respect to a fresh
+DKG phase. It means, if a trust party gathered a threshold of private shares, it
+could generate any beacon of the chain. When a resharing occurs, the individual
+shares of each drand node change but the chain remains the same as well. If the
+same set of nodes perform a new fresh DKG, it will create a new chain from
+scratch.
+
+**Resharing**: During a resharing, a drand network A (with threshold
+`tA`)creates new shares for a drand network B (with threshold `tB`) which can be
+disjoinct from A, such the drand network B is now responsible to continue
+creating drand beacons, while that the distributed public key doesn't change.
+For this to happen, there needs to be at least `tA` nodes from network A and
+`tB` nodes online and honest during the resharing. At the end of the protocol,
+there are going to be at least `tB` nodes that are qualified and have private shares
+to generate randomness.
+
+## Attacks
+
+### Randomness Generation
+
+There can be multiple ways of attacking the drand network during the randomness
+generation phase, each with different consequences.
+
+#### Front Running 
+
+**Scenario**: An attacker that is able to listen passively on the traffic
+between nodes (if TLS is not used - which is not a recommened setup) OR that is
+able to listen to plaintext traffic from the network of a threshold of nodes might be
+able to see a threshold number of partial beacons before any other honest nodes.
+
+**Consequence**: The attacker in such position is able to aggregate the final
+beacon of the current round before any other drand nodes. The attacker doesn't
+have to submit the beacon to the relay node and can already use the beacon for
+the application that uses drand. The delay between the time the attacker has the
+final beacon and the other nodes have it depends on the network connectivity
+between the drand nodes. Under normal circumstances, given a global passive listener
+attacker (which is arguably a strong model), the delay between the time the
+attacker computes the final beacon and the application's end users receive it
+from the relay network is on the order of hundreds of ms: the time that at least
```

Comment:
hundreds of ms is a lot! i'd hope for 10's

Reply:
I know i shouldn't have put a concrete number :D 
TBH I have no idea of what this number could be, it really depends on the connectivity of the drand nodes, to the drand relay and connectivity of the enduser to the drand relay, and also depends on the capacity of the attacker to be at every outgoing edges of every node... 
I just thought of the worse number possible but in practice under regular network environment, it could be much lower than that. 

I'll rephrase I think without giving specific number.

Reply:
Sounds good. i think talking in terms of wide-area round-trip-times when it's purely racing, and then alternatively of how many edges would need to be deactivated for the attacker to learn the value without the  group getting to the public value is maybe the easiest way to frame it.